### PR TITLE
fixes #4658 - parse puppet.conf with augeas instead of puppet internals

### PR DESCRIPTION
**Pull request by domcleal**
@ares?

Tested on EL6 with Puppet 3.5 successfully.

### CODE REVIEW

**Comment by domcleal** [comment]
[test]

File: lib/proxy/puppet/config_reader.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        begin
+          aug = Augeas::open(nil, nil, Augeas::NO_MODL_AUTOLOAD)
+          aug.set('/augeas/load/Puppet/lens', 'Puppet.lns')
+          aug.set('/augeas/load/Puppet/incl', config)
+          aug.load
+
+          @config_hash = {}
+          aug.match("/files#{config}/*/*[label() != '#comment']").each do |path|
+            (section, key) = path.split('/')[-2..-1]
+            (@config_hash[section.to_sym] ||= {})[key.to_sym] = aug.get(path)
```
Comment (ares) [inline lib/proxy/puppet/config_reader.rb]: I'd consider defining config_hash like this

``` ruby
@config_hash = Hash.new { |h, k| h[k] = {} }
```

so I could do this

``` ruby
@config_hash[section.to_sym][key.to_sym] = aug.get(path)
```

also I'd

``` ruby
(section, key) = path.split('/')[-2..-1].map(&:to_sym)
```

so result would look like

``` ruby
@config_hash[section][key] = aug.get(path)
```

Comment (domcleal) [inline lib/proxy/puppet/config_reader.rb]: Nice suggestions, updated.

**Comment by ares** [comment]
Code :+1:, untested though, should I try to install puppet nightlies or did you tested carefully?

**Comment by domcleal** [comment]
I've tested reasonably careful I think, patching a package installation and then doing a Puppet module import into Foreman successfully.

Just fixed the tests, I hope.

**Comment by ares** [comment]
Merged as dd37400544ff6f50bca77a1c13a82a3073cb8ebd, thanks @domcleal

**Pull request closed without merging**